scripts/actions/check-job-progress.js

```diff
@@ -100,6 +100,88 @@ const getBatchStatus = async ({ batchUid, jobId }) => {
   }
 };
 
+/**
+ * @param {Object[]} slugStatuses
+ * @param {Boolean} slugStatuses[].ok - Boolean flag signaling if translation deserialized. If it did, we can treat this translation as having completed.
+ * @param {String} slugStatuses[].slug - Slug representing file path translated.
+ * @param {String} slugStatuses[].jobId - Job id translation is associated with.
+ */
+const aggregateStatuses = (slugStatuses) => {
+  let totalFailures = 0;
+  let totalSuccesses = 0;
+  const jobStatuses = {};
+
+  slugStatuses.map(({ ok, jobId }) => {
+    if (!(jobId in jobStatuses)) {
+      jobStatuses[jobId] = { successes: 0, failures: 0 };
+    }
+
+    if (ok) {
+      totalSuccesses += 1;
+      jobStatuses[jobId].successes += 1;
+    } else {
+      totalFailures += 1;
+      jobStatuses[jobId].failures += 1;
+    }
+  });
+
+  return { totalFailures, totalSuccesses, jobStatuses };
+};
+
+/**
+ * @param {Object[]} slugStatuses
+ * @param {Boolean} slugStatuses[].ok - Boolean flag signaling if translation deserialized. If it did, we can treat this translation as having completed.
+ * @param {String} slugStatuses[].slug - Slug representing file path translated.
+ * @param {String} slugStatuses[].jobId - Job id translation is associated with.
+ */
+const updateTranslationRecords = async (slugStatuses) => {
+  // TODO: need to update this when we implement multiple locales. This only works for one locale.
+
+  await Promise.all(
+    slugStatuses.map(async ({ ok, slug }) => {
+      let updateStatus = ok ? 'COMPLETED' : 'ERRORED';
+
+      const record = await updateTranslations(
+        { slug, status: 'IN_PROGRESS' },
+        { status: updateStatus }
+      );
+
+      console.log(`Translation ${record.id} marked as ${updateStatus}`);
+    })
+  );
+};
+
+/**
+ * @param {Object.<string, Object.<string, number>} jobStatuses - object whose top level keys are job ids, and whose values are { succeses: x, failures: y }, which describe failure and success count for that given job.
+ */
+const updateJobRecords = async (jobStatuses) => {
+  await Promise.all(
+    Object.keys(jobStatuses).map(async (jobId) => {
+      const { successes, failures } = jobStatuses[jobId];
+
+      const records = await getTranslationsJobsRecords({
+        job_id: jobId,
+      });
+
+      if (successes + failures === records.length) {
+        if (failures > 0) {
+          await updateJob(jobId, { status: 'ERRORED' });
+          console.log(`Job ${jobId} marked as ERRORED`);
+        } else {
+          await updateJob(jobId, { status: 'COMPLETED' });
+          console.log(`Job ${jobId} marked as COMPLETED`);
+        }
+      } else {
```

Comment:
if 100 files were uploaded as part of Job A. and 1 file was canceled, but the job was eventually completed. when we go to check the job status and download the files, we _may_ have a mismatch between the database and smartling data, because we filter out files that had a status of `CANCELED` a la:

https://github.com/newrelic/docs-website/blob/00d3ba65717b68384ffa03fda2045871e9021bb7/scripts/actions/check-job-progress.js#L63